### PR TITLE
fix: navigation property will work even if they are not draft enabled

### PR DESCRIPTION
**Pull request by nlunets**
Internal incident
Bit difficult to test as CAP seems to enforce the draft status on navigation properties.
@drktfl any idea besides xml metadata ?

### CODE REVIEW

**Comment by nlunets** [comment]
This is an internal incident btw

**Comment by drktfl** [comment]
this will fail because the navigation to non-draft will still add the draft flags to the result set. We can probably ignore this? However, the same errors happen without the code change in dataAccess.ts, so probably we aren't testing what is relevant or did not fix anything 🤔

File: packages/fe-mockserver-core/src/data/dataAccess.ts [hunk 1]
```diff
@@ -276,14 +276,20 @@ export class DataAccess implements DataAccessInterface {
...
+                    (targetEntitySet?.annotations?.Common as any)?.DraftRoot ||
+                    (targetEntitySet?.annotations?.Common as any)?.DraftNode;
+                if (currentEntitySet && navigationData.hasOwnProperty('IsActiveEntity') && isTargetDraftEnabled) {
+                    currentKeys['IsActiveEntity'] = navigationData.IsActiveEntity;
+                } else if (
                     currentEntitySet &&
                     navigationData.hasOwnProperty('IsActiveEntity') &&
-                    ((currentEntitySet?.annotations?.Common as any)?.DraftNode ||
-                        (currentEntitySet?.annotations?.Common as any)?.DraftRoot)
+                    navPropDetail.targetType.keys.find((key: Property) => key.name === 'IsActiveEntity')
```
Comment (nlunets) [inline packages/fe-mockserver-core/src/data/dataAccess.ts:288]: Because of this line we only add the property if it is included in the target keys

**Comment by drktfl** [comment]
this indeed only works with XML metadata. The CDS compiler creates all sorts of constraints, but nothing really useful. They are either incomplete or missing, which makes it hard to write a predictable test.

I pushed a CDS model, but only the metadata.xml is useful for testing because of the manual adjustments needed to cover more cases than the standard "incomplete constraints" case that the compiler produces for 1:1 navigations, or the "no constraints" case it produces for 1:n.

I probably shouldn't have set the parameter type for the navigation properties along the way, it now bloated the change

**Pull request merged**